plugins/woocommerce/src/Internal/Utilities/FilesystemUtil.php

```diff
@@ -31,6 +32,26 @@ public static function get_wp_filesystem(): WP_Filesystem_Base {
 		return $wp_filesystem;
 	}
 
+	/**
+	 * Get the WP filesystem method, with a fallback to 'direct' if no FS_METHOD constant exists and there are not FTP related options/credentials set.
+	 *
+	 * @return string|false The name of the WP filesystem method to use.
+	 */
+	public static function get_wp_filesystem_method_or_direct() {
+		$proxy = wc_get_container()->get( LegacyProxy::class );
+		if ( ! $proxy->constant_is_defined( 'FS_METHOD' ) && false === $proxy->call_function( 'get_option', 'ftp_credentials' ) && ! $proxy->constant_is_defined( 'FTP_HOST' ) ) {
```

Comment:
Instead of checking if `ftp_credentials` is false, should we check if it is anything other than a populated array?

Just to be clear, I don't actually think we should do this, as it feels like a private implementation detail we shouldn't be concerned about, but I decided to leave a note in any case so you could double check my thinking 🙃